README.md
# `@aeries-design/adl-tailwind-config`

## Installation

### Default Process

Tailwind provides good installation instructions on how to get setup with their tools, which you can find [here][tailwind-installation].

With imports from Tailwind CSS in place, you can leverage the ADL configuration like this:

```javascript
// tailwind.config.js
module.exports = {
  presets: [require('@aeries-design/adl-tailwind-config')],
};
```

### Pre-Built File

In situations where you do not want to setup the build pipeline  uses a command-line tool to build the css from source, a pre-built version of the ADL Tailwind css file is included in this package. **NOTE** This method will use the Tailwind CSS file in it's entirety and will not do any tree-shaking to purge unused classes.

### From your dependencies

For tools that allow you to include a pre-existing CSS file into your application, this package includes a pre-built version of the CSS file that you can import. For example, if you're using `gridsome create $project-name`, you could add the following to your `src/index.js`

```javascript
import '@aeries-design/adl-tailwind-config/dist/adl-tailwind.css';
// or the minified version
import '@aeries-design/adl-tailwind-config/dist/adl-tailwind.min.css';
```
and the tailwind classes will be made available to you.

#### From a CDN

If you want to go even _simpler_, you can link against the pre-built CSS file on [`unpkg`][unpkg]. Include the following in the `head` of your document.

```html
<link 
  rel="stylesheet" 
  href="https://unpkg.com/@aeries-design/adl-tailwind-config/dist/adl-tailwind.min.css"
>
```
Note that with this approach you will _always_ get the latest version of Tailwind CSS -- even if there was a major version bump. See the `unpkg` documentation if you want to link against a specific version of the package.

### Fonts

Regardless of which implementation method you choose to use, you _also_ must link to the font definitions we use as part of the ADL.


#### Google Fonts CDN

include the following `link` in the `head` of your document, _before_ any other CSS files.

```html
<link 
  rel="preconnect" 
  href="https://fonts.gstatic.com"
>
<link
  rel="stylesheet"
  href="https://fonts.googleapis.com/css2?family=Nunito+Sans:ital,wght@0,200;0,300;0,400;0,600;0,700;0,800;1,200;1,300;1,400;1,600;1,700;1,800&display=swap" 
>
```

[tailwind]: https://tailwindcss.com
[tailwind-installation]: https://tailwindcss.com/docs/installation
[unpkg]: https://unpkg.com


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [1.0.7](https://github.com/aeries-design/tailwind-config/compare/v1.0.6...v1.0.7) (2022-12-21)

### [1.0.6](https://github.com/aeries-design/tailwind-config/compare/v1.0.5...v1.0.6) (2022-01-10)

### [1.0.5](https://github.com/aeries-design/tailwind-config/compare/v1.0.3...v1.0.5) (2021-12-03)

### [1.0.3](https://github.com/aeries-design/tailwind-config/compare/v1.0.2...v1.0.3) (2021-08-06)

### [1.0.2](https://github.com/aeries-design/tailwind-config/compare/v1.0.1...v1.0.2) (2021-04-14)


### Bug Fixes

* Add util directory to npm package files. ([00d32d9](https://github.com/aeries-design/tailwind-config/commit/00d32d9ecdeebebab284cb3f1ab5f1e6aabccccf))

### 1.0.1 (2021-04-14)


### Bug Fixes

* remove pxToRem function from screens so screen definitions compile correctly. ([19e7dd9](https://github.com/aeries-design/tailwind-config/commit/19e7dd940bd2619c6ca85dd49e64bf53d6f613f1))


index.js
'use strict';

const { ...colors } = require('./config/colors');
const spacing = require('./config/spacing');
const {
  boxShadow,
  fontFamily,
  fontSize,
  fontWeight,
  letterSpacing,
  maxHeight,
  maxWidth,
  minHeight,
  zIndex,
} = require('./config/typography');

const variants = require('./config/variants');
const { gridRow, gridColumn } = require('./config/grid');
const screens = require('./config/screens');

module.exports = {
  theme: {
    colors,
    container: { center: true },
    fill: {
      ...colors,
      current: 'currentColor'
    },
    fontFamily,
    fontWeight,
    screens,
    stroke: {
      ...colors,
      current: 'currentColor'
    },
    extend: {
      boxShadow,
      fontSize,
      gridRow,
      gridColumn,
      letterSpacing,
      maxHeight,
      maxWidth,
      minHeight,
      spacing,
      zIndex,
    }
  },
  variants,
  corePlugins: {
    translate: false,
  }
}

config/colors.js
'use strict';

/** Tailwind treates a key called `default` as the value to use
 * when a color is accessed without a nested specifier. 
 * (ie. blue instead of blue-400)
 */

const addDefault = (colorDefinition, defaultKey = '500') => {
  return {
    DEFAULT: colorDefinition[defaultKey],
    ...colorDefinition
  };
}

const neutral = addDefault(
  {
    '2000':       '#060E1E',
    '1900':       '#0C1E41',
    '1800':       '#162B4D',
    '1700':       '#203858',
    '1600':       '#344563',
    '1500':       '#41526E',
    '1400':       '#505F79',
    '1300':       '#5E6D85',
    '1200':       '#6C788D',
    '1100':       '#7A879B',
    '1000':       '#8993A4',
    '900':        '#96A0AF',
    '800':        '#A5ADBA',
    '700':        '#B2BAC5',
    '600':        '#C0C6D0',
    '500':        '#DFE3E9',
    '400':        '#ECEEF2',
    '300':        '#F6F8FA',
    '200':        '#FCFDFE',
    '100':        '#ffffff'
  },
  '1300'
);

const purple = addDefault(
  {
    '800':        '#362C83',
    '700':        '#3F3894',
    '600':        '#54489E',
    '500':        '#6359A7',
    '400':        '#8076B6',
    '300':        '#9A8CC3',
    '200':        '#C0B5D9',
    '100':        '#EBE6F2'
  }
);

const orange = addDefault(
  {
    '800':        '#C55B18',
    '700':        '#F36B21',
    '600':        '#F6891E',
    '500':        '#FAA31A',
    '400':        '#FFD503',
    '300':        '#FCF58B',
    '200':        '#F7F2C0',
    '100':        '#F9F6EB'
  }
);

const green = addDefault(
  {
    '800':        '#0F5636',
    '700':        '#0F6745',
    '600':        '#068A5B',
    '500':        '#34B67F',
    '400':        '#73C7A0',
    '300':        '#9FD3B9',
    '200':        '#C2E1CE',
    '100':        '#E7F4EB'
  }
);

const red = addDefault(
  {
    '800':        '#8E0F1B',
    '700':        '#BE212F',
    '600':        '#D43027',
    '500':        '#E64C33',
    '400':        '#EE6A51',
    '300':        '#EB806B',
    '200':        '#F0AFA3',
    '100':        '#F3DCD6'
  },
  '700'
);

const blue = addDefault(
  {
    '800':        '#193878',
    '700':        '#1649A0',
    '600':        '#345BA9',
    '500':        '#396DB5',
    '400':        '#467FC1',
    '300':        '#57A5DA',
    '200':        '#B6E3FA',
    '100':        '#D4EAF8',
  },
  '600'
);

const white = '#FFFFFF';
const black = '#060E1E';

module.exports = {
  neutral,
  red,
  blue,
  green,
  orange,
  purple,
  white,
  black,
  current: 'currentColor'
}

config/typography.js
'use strict';

const pxToRem = require('../util/pxToRem')

const fontFamily = {
  'aeries':   ['Aeries Sans'],
  'nunito' :  ['Nunito Sans']
}

const fontWeight = {
  thin        : 200,
  light       : 300,
  medium      : 400,
  'semi-bold'   : 600,
  bold        : 700,
  'extra-bold'  : 800
}

const fontSize = {
  'xx-large'    : '3.06rem',
  'display'     : '2.43rem',
  'header'      : '1.93rem',
  'title'       : '1.56rem',
  'subhead'     : '1.15rem',
  'body'        : '1rem',
  'minimum-text': '.81rem',
  'fine-print'  : '.625rem'
}

const zIndex = {
  '-10': -10,
  '-5': -5,
  '-1': -1,
  '5': 5,
  '15': 15,
  '25': 25,
  '35': 35,
  '45': 45,
  '55': 55,
  '60': 60,
  '65': 65,
  '70': 70,
  '75': 75,
  '80': 80,
  '85': 85,
  '90': 90,
  '95': 95,
  '100': 100,
}

const maxWidth = {
  '32':   '8rem',
  '56':   '14rem',
  'page-inset': '56rem',
  '1/2': '50%',
  '1/4': '25%',
  '3/4': '75%',
  '1/3': '33.33333%',
  '2/3': '66.66667%',
  '1/5': '20%',
  '2/5': '40%',
  '3/5': '60%',
  '4/5': '80%',
  '1/6': '16.666667%',
  '5/6': '83.333333%',
  '960': '60rem'
}

const minHeight = {
  '24': '6rem',
  '32': '8rem',
}

const maxHeight = {
  '0': pxToRem('0'),
}

const boxShadow = {
  'double-drop': '0px 4px 25px rgba(22, 43, 77, 0.11), 0px 4px 8px rgba(94, 109, 133, 0.04)'
}

const letterSpacing = {
  'superwide': '.75rem',
}

module.exports = {
  boxShadow,
  fontFamily,
  fontSize,
  fontWeight,
  letterSpacing,
  maxHeight,
  maxWidth,
  minHeight,
  zIndex,
}